_posts/2024-01-25-websockets-in-http4s.md

```diff
@@ -0,0 +1,1561 @@
+---
+title: "WebSockets in Http4s"
+date: 2024-01-25
+header:
+  image: "/images/blog cover.jpg"
+tags: []
+excerpt: "An Introduction to WebSockets In Http4s."
+toc: true
+toc_label: "In this article"
+---
+
+_by [Herbert Kateu](https://github.com/hkateu)_
+
+## 1. Introduction
+
+The WebSocket protocol enables persistent two-way communication between a client and a server where packets can be passed in both directions without the need for additional HTTP requests.
+
+The specification for this protocol is outlined in [RFC 6455](https://datatracker.ietf.org/doc/html/rfc6455). WebSockets are used in applications such as Instant Messaging, Gaming, Simultaneous editing, and stock tickers to mention but a few.
+
+In this article, we'll be creating a chat application using Http4s' implementation of webSockets.
+
+## 2. Setting Up
+
+### 2.1. Updating build.sbt
+
+We'll set up our project using the Http4s `giter8` quick start template by running the following code:
```

Comment:
Is the g8 template any useful, since we don't need most of it and we need to explain the server anyway?

Reply:
Removed g8 bit and edited the build.sbt